### PR TITLE
Issues on binary = error, issues on library = warning for BA2004

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs [hunk 1]
```diff
@@ -134,6 +142,22 @@ public void AnalyzeNativeBinaryAndPdb(BinaryAnalyzerContext context)
...
+        {
+            string compilands = compilandsWithOneOrMoreInsecureFileHashes.CreateOutputCoalescedByLibrary();
+
+            //'{0}' is a native binary that links one or more object files which were hashed
+            // using an insecure checksum algorithm (MD5). MD5 is subject to collision attacks
+            // and its use can compromise supply chain integrity. Pass '/ZH:SHA-256' on the
+            // cl.exe command-line to enable secure source code hashing. The following modules
+            // are out of policy: {1} 
+            context.Logger.Log(this,
+                RuleUtilities.BuildResult(FailureLevel.Warning, context, null,
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs:155]: >Warning [](start = 55, length = 7)

Almost done! You need to pass the failure level into this helper, Error for binary compilands, Warning for libray. :) Then see what test breaks occur. Run UpdateBaselines.ps1 in the driver functionals directory and review changes.

Comment (eddynaka) [reply src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs:155]: oops after i create the function, i forgot to add the level. thanks

---
In reply to: [554458619](https://github.com/microsoft/binskim/pull/328#discussion_r554458619) [](ancestors = 554458619)

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif [hunk 2]
```diff
@@ -638,10 +662,10 @@
                   "text": "'{0}' was not evaluated for check '{1}' as the analysis is not relevant based on observed metadata: {2}."
                 }
               },
-              "shortDescription": {
-                "text": "Binaries should be marked as high entropy Address Space Layout Randomization (ASLR) compatible."
-              },
               "name": "EnableHighEntropyVirtualAddresses",
+              "defaultConfiguration": {
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif:666]: >defaultConfiguration [](start = 15, length = 20)

this looks bad! this is not a legal value to set this! did the new SARIF SDK introduce this change?

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif [hunk 3]
```diff
@@ -604,10 +628,10 @@
                   "text": "'{0}' is a kernel mode portable executable compiled for a version of Windows that does not support the control flow guard feature for kernel mode binaries."
                 }
               },
-              "shortDescription": {
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Native_x64_VS2013_Default.dll.sarif:607]: >shortDescription [](start = 15, length = 16)

so, we have James' change! Nice! looks good.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**